tensorflow/python/keras/metrics.py

```diff
@@ -870,6 +870,30 @@ class SparseCategoricalAccuracy(MeanMetricWrapper):
   def __init__(self, name='sparse_categorical_accuracy', dtype=None):
     super(SparseCategoricalAccuracy, self).__init__(
         sparse_categorical_accuracy, name, dtype=dtype)
+  
+  def update_state(self, y_true, y_pred, sample_weight=None):
+    """Accumulates metric statistics.
+
+    The shapes of `y_true` and `y_pred` are different.
+
+    Args:
+      y_true: Ground truth label values. shape = `[batch_size, 1]`.
```

Comment:
It doesn't necessary to be [batch_size, 1], it should still have the same rank as the y_pred, just the last dim should be 1.

Reply:
How about,
```suggestion
      y_true: Ground truth label values. shape = `[batch_size, d1, .. dN-1]` or shape = `[batch_size, d1, .. dN-1, 1]`.
```

The rank of `y_true` can be short from the rank of `y_pred` by 1. Or, the ranks can be the same.
It is from the sparse_categorical_accuracy(y_true, y_pred) function at
https://github.com/ziruizhuang/tensorflow/blob/d6ee4a941b79a464a9b229f6a22be50baab62d4d/tensorflow/python/keras/metrics.py#L3514-L3517